### PR TITLE
first test for webWriter + .github changes + serverInfo struct w/ methods

### PR DESCRIPTION
**Pull request by gauron99**
* created new tests for `webWriter()` 
* gh workflow running for basic tests 
@zroubalik

### CODE REVIEW

File: http-server/server/server.go [hunk 1]
```diff
@@ -11,17 +12,31 @@ import (
...
 
 var pageAccessCount = 0
-var Param string
+var param string = ""
 
 var greetings = []string{"Hello", "Greetings", "Welcome", "Hi"}
 var titles = []string{"the Mighty Traveler", "the Great Summoner", "the Conqueror of Titans", "the Destroyer of Worlds", "the King of Oceans", "the King of Underworld"}
 
+// GetParam returns pointer to param variable (for other packages)
+func GetParam() *string {
```
Comment (zroubalik) [inline http-server/server/server.go:21]: Would be better to return plain string, not pointer, see: https://dhdersch.github.io/golang/2016/01/23/golang-when-to-use-string-pointers.html

File: http-server/server/server.go [hunk 2]
```diff
@@ -11,17 +12,31 @@ import (
...
-func WebWriter(rw http.ResponseWriter, s string) {
+func webWriter(rw http.ResponseWriter, s string) {
 	fmt.Fprintf(rw, "%s\n", s)
-	if len(Param) > 0 { //if a parameter is given, print it out everywhere where something is printed out
-		fmt.Fprintf(rw, "I have a parameter! Here: %s\n", Param)
+	if len(param) > 0 { //if a parameter is given, print it out everywhere where something is printed out
+
+		out := fmt.Sprintf("I have a parameter! Here: %s\n", param)
+		// outPrt := fmt.Sprintf("I have a parameter!! Here: %s\n", param)
+		// fmt.Fprintf(rw, outPrt)
```
Comment (zroubalik) [inline http-server/server/server.go:38]: is this needed?

File: http-server/server/server.go [hunk 3]
```diff
@@ -11,17 +12,31 @@ import (
 )
 
 var pageAccessCount = 0
-var Param string
+var param string = ""
```
Comment (zroubalik) [inline http-server/server/server.go:15]: Would be better to convert this to a struct:
```go
type server struct (
  param string
  pageAccessCount int
}
```

And then create a methods on this struct.

File: http-server/server/server_test.go [hunk 4]
```diff
@@ -5,47 +5,86 @@ import (
...
+// 	}
+
+// 	expected := time.Now().Format("2006-01-02 15:04:05")
+// 	got := rr.Body.String()
+
+// 	fmt.Println("exp:", expected, "; got:", got)
+// 	if got != expected {
+
+// 		t.Errorf("Http body fail. Expected %v but got %v ", expected, got)
+// 	}
```
Comment (zroubalik) [inline http-server/server/server_test.go:89]: In general it is not a good practice to commit blocks of commented code.

File: http-server/server/server.go (hunk 1 above)
Comment (gauron99) [reply http-server/server/server.go]: rewrote using plain strings

**Comment by gauron99** [comment]
@zroubalik 
* rewrote server parameters into `struct`
* updated `GetParam` to be a method for this struct using plain string instead of pointer to string

File: http-server/server/server.go [hunk 5]
```diff
@@ -11,29 +11,34 @@ import (
...
 
-var pageAccessCount = 0
-var param string = ""
+// change this to use methods in the file
+type serverInfo struct {
+	pageAccessCount int
+	param           string
+}
+
+var serv = serverInfo{}
```
Comment (gauron99) [inline http-server/server/server.go:20]: `var Serv = serverInfo{}`
possibly have this public in order to use in `main` as `server.Serv`
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

File: http-server/server/server.go [hunk 6]
```diff
@@ -11,29 +11,34 @@ import (
...
 
 var greetings = []string{"Hello", "Greetings", "Welcome", "Hi"}
 var titles = []string{"the Mighty Traveler", "the Great Summoner", "the Conqueror of Titans", "the Destroyer of Worlds", "the King of Oceans", "the King of Underworld"}
 
-// GetParam returns pointer to param variable (for other packages)
-func GetParam() *string {
-	return &param
+// SetParam takes string argument and sets it as parameter for the program
+func SetParam(s string) {
```
Comment (gauron99) [inline http-server/server/server.go:26]:
```
func (si *serverInfo) SetParam(s string) {
	si.param = s
}
```
reference the above comment

File: http-server/server/server.go [hunk 7]
```diff
@@ -10,18 +11,37 @@ import (
...
 
-var pageAccessCount = 0
-var Param string
+// change this to use methods in the file
+type serverInfo struct {
+	pageAccessCount int
+	param           string
+}
+
+var serv = serverInfo{}
```
Comment (gauron99) [inline http-server/server/server.go:20]: `var Serv = serverInfo{}`
possibly have this public in order to use in main as `server.Serv`
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

File: http-server/server/server.go (hunk 5 above)
Comment (gauron99) [inline http-server/server/server.go:20]: `var Serv = serverInfo{}`
possibly have this public in order to use in main as server.Serv
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

File: http-server/server/server.go [hunk 8]
```diff
@@ -10,18 +11,37 @@ import (
...
 
 var greetings = []string{"Hello", "Greetings", "Welcome", "Hi"}
 var titles = []string{"the Mighty Traveler", "the Great Summoner", "the Conqueror of Titans", "the Destroyer of Worlds", "the King of Oceans", "the King of Underworld"}
 
+// SetParam takes string argument and sets it as parameter for the program
+func SetParam(s string) {
+	serv.param = s
+}
```
Comment (gauron99) [inline http-server/server/server.go:28]:
```
func (si *serverInfo) SetParam(s string) {
	si.param = s
}
```
reference the comment above

File: http-server/server/server.go (hunk 2 above)
Comment (gauron99) [reply http-server/server/server.go]: removed with latest commit

**Comment by gauron99** [comment]
* made SetParam() as a method
* removed some useless comments

File: http-server/main.go [hunk 9]
```diff
@@ -27,13 +27,13 @@ func main() {
 
 	// handle possible cli arguments & set
 	flag.Parse()
-	server.Serv.SetParam(paramPtr)
+	serv := server.MakeServerInfo(0, "")
```
Comment (zroubalik) [inline http-server/main.go:30]: rename to `NewServerInfo` it's more idiomatic

File: http-server/main.go [hunk 10]
```diff
@@ -27,13 +27,13 @@ func main() {
...
+	serv.Param = paramPtr
```
Comment (zroubalik) [inline http-server/main.go:31]: Why like this and not part of the function above?

File: http-server/server/server.go [hunk 11]
```diff
@@ -14,39 +14,32 @@ import (
 // change this to use methods in the file
 type serverInfo struct {
 	pageAccessCount int
-	param           string
+	Param           string
```
Comment (zroubalik) [inline http-server/server/server.go:17]: this don't have to be public, if you initiate it in contructor.

**Review by zroubalik — COMMENTED** [review]
almost there :) No need to treat Param differently.

File: http-server/server/server.go [hunk 12]
```diff
@@ -10,25 +11,35 @@ import (
...
-var pageAccessCount = 0
-var Param string
+// change this to use methods in the file
+type serverInfo struct {
+	pageAccessCount int
+	Param           string
+}
+
+// MakeServerInfo creates & returns newly initialized serverInfo structure with given args
+func MakeServerInfo(pageAccessCnt int, parameter string) *serverInfo {
```
Comment (zroubalik) [inline http-server/server/server.go:21]: It kinda doesn't make a sense to set pageAccessCnt here via parameter.
Just set it init it in the constructor without param.

**Review by zroubalik — APPROVED** [review]
LGTM

**Review by zroubalik — APPROVED** [review]
LGTM :) ready to merge

**Pull request merged**